cpp/src/arrow/array.h

```diff
@@ -522,6 +522,44 @@ class ARROW_EXPORT ListArray : public Array {
   std::shared_ptr<Array> values_;
 };
 
+// ----------------------------------------------------------------------
+// FixedSizeListArray
+
+/// Concrete Array class for list data
+class ARROW_EXPORT FixedSizeListArray : public Array {
```

Comment:
is it worth having common abstract class?

Reply:
There's not much opportunity for code reuse in arrays/builders. The only advantage I can think of would be easier reuse of code which consumes `ListArray` (but only code which doesn't directly access the offsets buffer). Those algorithms could use `ArrayData BasicListArray::GetView(int64_t i)` or similar. This seems like speculative generality to me; I'd be more open to doing it if you can think of existing code which would benefit from being refactored this way.

`FixedSizeListScalar` and `ListScalar` are identical but also trivial.

Reply:
Agreed with @bkietz. Also code reuse can be achieved through templating (we generally care about performing non-virtual calls anyway).

Reply:
:+1: